### ISSUE TITLE
is_duration raises an uncaught decimal.Overflow on a duration with a large exponent

### ISSUE THREAD

**Issue by vtushar06** [python-jsonschema/jsonschema#1511]
The `duration` format check can raise `decimal.Overflow` instead of returning a result, so validating certain strings throws an uncaught exception rather than reporting the instance as invalid.

### Versions

- jsonschema 4.25.1
- isoduration 20.11.0 (the optional dependency `is_duration` delegates to)
- Python 3.9

### Reproduction

```python
from jsonschema import Draft202012Validator as V
from jsonschema import FormatChecker

v = V({"format": "duration"}, format_checker=FormatChecker())
list(v.iter_errors("P1E1000000D"))
```

raises:

```
decimal.Overflow: [<class 'decimal.Overflow'>]
```

### Cause

`_format.is_duration` calls `isoduration.parse_duration(instance)` and is registered with `raises=isoduration.DurationParsingException`. isoduration parses each component amount with `Decimal(...)`, and for an exponent past the decimal context's `Emax` (999999), `Decimal("1E1000000")` raises `decimal.Overflow`. That is not a subclass of `DurationParsingException`, so it propagates out of the format checker uncaught.

The boundary is exact: `P1E999999D` is accepted, `P1E1000000D` overflows. It is also reachable without an exponent, via a plain digit run longer than `Emax`.

### Expected

The check should report the instance as invalid (`P1E1000000D` is not a valid RFC 3339 Appendix A duration - the grammar uses `1*DIGIT`, with no exponent), not raise.

### Possible fix

Either broaden what `is_duration` catches (add `ArithmeticError` / `decimal.Overflow` alongside `DurationParsingException`, or wrap the `parse_duration` call), or treat any exception from `parse_duration` as "not a duration".

**Comment by mathewOracle** [comment]
I ran into this too and dug in a bit. Root cause: `FormatChecker.check()` only converts exceptions in a format's `raises=` tuple into "invalid instance" - `duration` only lists `isoduration.DurationParsingException`, but `isoduration.parse_duration` uses `Decimal(...)` internally, which raises `decimal.Overflow` for a magnitude/exponent past `Emax`. So it escapes uncaught instead of just reporting the instance as invalid.

Fix is small: add `decimal.DecimalException` (the common parent of `Overflow` and friends) alongside `DurationParsingException` in that tuple. Added a regression test too - fails on the current code with the exact traceback from this issue, passes with the fix. Full suite (374 tests) still green, ruff clean.

I can't open a PR directly (repo has PR creation limited to collaborators), so it's sitting on my fork if useful: https://github.com/mathewOracle/jsonschema/compare/main...mathewOracle:jsonschema:fix/1511-duration-decimal-overflow

Happy to also raise a case against `bolsote/isoduration` if that's the more correct home for it, given the "Needs Test Upstream" label - let me know which you'd prefer.
